src/mono/mono/mini/mini-arm64.c

```diff
@@ -5306,6 +5425,40 @@ mono_arch_emit_exceptions (MonoCompile *cfg)
 		set_code_cursor (cfg, code);
 	}
 
+	/* Handle relocations with RIP relative addressing */
+	for (ji = cfg->patch_info; ji; ji = ji->next) {
+		gboolean remove = FALSE;
+
+		if (ji->type == MONO_PATCH_INFO_X128) {
+			guint8 *pos, *patch_pos;
+			guint32 target_pos;
+
+			code = (guint8*)ALIGN_TO (code, 16);
+			pos = cfg->native_code + ji->ip.i;
+			patch_pos = pos + 3;
+			target_pos = GPTRDIFF_TO_UINT32 (code - pos - 4);
+			memcpy (code, ji->data.target, 16);
+			code += 16;
+
+			*(guint32*)(patch_pos) = target_pos;
```

Comment:
Looks like this should overwrite the address in a `b` or `bl` instruction to whatever is in `target_pos`. So I'd use `patch_pos = pos` and change `*(guint32*)(patch_pos)...` to `arm_bl (patch_pos, target_pos);`. This should correctly replace the branch. Though the patch points seem to be a little more complicated (using thunks if displacements are large), I'm not sure if we should account for those?